New package: Kitware.ParaView.MPI version 5.11.2

Dragon1573 opened the pull request:
## What's changed

This is an MPI accelerated version of `Kitware.ParaView`.

## Checks

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/122402)

Dragon1573 committed:
New package: Kitware.ParaView.MPI version 5.11.2

Dragon1573 committed:
Fixup: "InstallerUrl"

- Forget to switch back the official URL
- Use a private "cache" URL for local test only

stephengillie commented:
Hi @Dragon1573,

The package installed normally, but failed to launch in a Windows 10 VM:
![image](https://github.com/microsoft/winget-pkgs/assets/114952565/b51964b1-ec97-4056-8a41-0f43c6fdc7f1)

This appears to be a dependency on [Microsoft MPI](https://www.microsoft.com/en-us/download/details.aspx?id=57467), which doesn't appear to have a public manifest yet.

stephengillie commented on manifests/k/Kitware/ParaView/MPI/5.11.2/Kitware.ParaView.MPI.installer.yaml:
```diff
@@ -0,0 +1,17 @@
+Scope: machine
+ProductCode: '{76D57FB1-0CD5-40A2-9296-16B85344BCAA}'
+AppsAndFeaturesEntries:
+- UpgradeCode: '{E06445A7-B257-4FCE-9241-2A189AD26B5A}'
```
```suggestion
- UpgradeCode: '{E06445A7-B257-4FCE-9241-2A189AD26B5A}'
Dependencies:
  PackageDependencies:
  - PackageIdentifier: Microsoft.MPI
```
Once the Issue is resolved, this change should resolve the application error. (Before the issue is resolved, this should cause a slightly different error due to the missing dependency.)

stephengillie commented:
Unfortunately, with MS-MPI preinstalled, the application crashes on launch. 

Manual Validation ended with:
> Faulting application name: paraview.exe, version: 0.0.0.0, time stamp: 0x6511a217  
> Faulting module name: unknown, version: 0.0.0.0, time stamp: 0x00000000 Exception code: 0xc0000005 
Fault offset: 0x0000000000000000  
> Faulting process id: 0x9e0  
> Faulting application start time: 0x01d9fc9e5a1376a4  
> Faulting application path: C:\Program Files\ParaView 5.11.2\bin\paraview.exe  
> Faulting module path: unknown Report Id: b446f8ae-ac45-488d-a595-920bb8915780  
> Faulting package full name:   
> Faulting package-relative application ID:

Dragon1573 commented:
> Unfortunately, with MS-MPI preinstalled, the application crashes on launch.

Does it mean `Kitware.ParaView.MPI` still have something unusual and cannot be accepted by `winget-pkgs`?

If this problem is still there and could not be resolved, I think we can close this pull request. `Kitware.ParaView` (without MPI acceleration) is already available in `winget-pkgs`.

stephengillie commented:
It turns out that MS-MPI already exists, as `Microsoft.msmpi`. Sorry for not checking more thoroughly.

stephengillie commented:
I've updated the suggestion above. If you will commit it then the pipeline can retry.

Dragon1573 committed:
Fixup: Add required dependencies

- It requires MSMPI for running MPI accelerated version
- Add `Microsoft.msmpi` as dependency

Co-authored-by: Stephen Gillie <114952565+stephengillie@users.noreply.github.com>
